### ISSUE TITLE
Make protoBufferPool public

### ISSUE THREAD

**Issue by vitalyisaev2** [grpc/grpc-go#1644]
It would like to use your implemenation of `*proto.Buffer` `sync.Pool`. Would you accept PR that makes this variable public? Or a fabric that returns `grpc.Codec` instance with a `protoBufferPool` inside.

**Comment by dfawley** [comment]
We plan on making some changes here in the next month or two that might make it unsuitable for export right now.  What exactly do you want this for?

FWIW, after #1547 gets hashed out, it will be possible to retrieve a grpc.Codec for the proto codec via [`encoding.GetCodec`](https://github.com/grpc/grpc-go/pull/1547/files#diff-c84dca0c336760ab46b754cffd381cd9R68).

**Comment by vitalyisaev2** [comment]
@dfawley thanks for a response, I just wanted to use your ```sync.Pool``` with ```*proto.Pool``` in a third-party project, and I didn't want to copy-paste code with tests from `grpc-go` :) 

Since you're planning to change it, please take into account, that it would be nice to have this structure (or maybe a method that initializes it) public. Thank you.

**Comment by dfawley** [comment]
FWIW, you are free to use the code in your project as long as you comply with the terms of the license at the top.

**Comment by vitalyisaev2** [comment]
@dfawley thank you, yes, it's possible... But if I copied this code, I'd have to maintain it (including tests), whereas these efforts have been already made by GRPC authors.

So I just wanted to notice that this code seem to be separable from the main code base, and it would be very nice to these structures public some day.
